spinalcordtoolbox/scripts/sct_image.py

```diff
@@ -220,7 +235,8 @@ def main(argv=None):
     if arguments.o is not None:
         fname_out = arguments.o
     else:
-        fname_out = None
+        # in case fname_out is not defined, use first element of input file name list
+        fname_out = fname_in[0]
```

Comment:
NB: This change is (kind-of) a bugfix/refactoring (https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3865/commits/916ef2e271ba8dbc1cf4dc2e5f3546576bfb6dd9), and could possibly be merged outside of this PR instead.